### PR TITLE
Bug OCPBUGS-1352: [DownstreamMerge] 11-18-22

### PR DESCRIPTION
**Pull request by trozet**
Had to include a carry commit to fix a hybrid overlay unit test failure that occurs downstream only

### CODE REVIEW

**Comment by trozet** [comment]
@dcbw @jcaamano

**Comment by trozet** [comment]
hmm seeing this from the dualstack failure:
```
event happened 243 times, something is wrong: node/worker-1.ostest.test.metalkube.org - reason/ErrorReconcilingNode roles/worker (combined from similar events): failed to create or update ACL &{5ec7be0d-aa44-4bcf-a969-0feb6d738850 allow-related to-lport map[] 0 false ip6.src==fd01:0:0:6::2 0xc000310310 <nil> map[] 1001 <nil>}: unexpectedly found multiple results for provided predicate
event happened 55 times, something is wrong: node/worker-1.ostest.test.metalkube.org - reason/ErrorReconcilingNode roles/worker (combined from similar events): failed to create or update ACL &{f3cb773d-0d25-46b6-8884-7920699fac3d allow-related to-lport map[] 0 false ip6.src==fd01:0:0:6::2 0xc0037532b0 <nil> map[] 1001 <nil>}: unexpectedly found multiple results for provided predicate
event happened 184 times, something is wrong: node/worker-1.ostest.test.metalkube.org - reason/ErrorReconcilingNode roles/worker (combined from similar events): failed to create or update ACL &{5ec7be0d-aa44-4bcf-a969-0feb6d738850 allow-related to-lport map[] 0 false ip6.src==fd01:0:0:6::2 0xc003205060 <nil> map[] 1001 <nil>}: unexpectedly found multiple results for provided predicate
event happened 178 times, something is wrong: node/worker-2.ostest.test.metalkube.org - reason/ErrorReconcilingNode roles/worker (combined from similar events): error creating gateway for node worker-2.ostest.test.metalkube.org: failed to init shared interface gateway: error creating static route {UUID:70449439-0373-49c3-b544-1424e1d6e867 BFD:<nil> ExternalIDs:map[] IPPrefix:0.0.0.0/0 Nexthop:192.168.111.1 Options:map[] OutputPort:0xc000630af0 Policy:<nil> RouteTable:} in GR GR_worker-2.ostest.test.metalkube.org: unexpectedly found multiple results for provided predicate
event happened 194 times, something is wrong: node/worker-2.ostest.test.metalkube.org - reason/ErrorReconcilingNode roles/worker (combined from similar events): error creating gateway for node worker-2.ostest.test.metalkube.org: failed to init shared interface gateway: error creating static route {UUID:0c5b6f24-0f1c-4c04-87a6-f40dc6fc3386 BFD:<nil> ExternalIDs:map[] IPPrefix:0.0.0.0/0 Nexthop:192.168.111.1 Options:map[] OutputPort:0xc0021d7030 Policy:<nil> RouteTable:} in GR GR_worker-2.ostest.test.metalkube.org: unexpectedly found multiple results for provided predicate
event happened 57 times, something is wrong: node/worker-2.ostest.test.metalkube.org - reason/ErrorReconcilingNode roles/worker (combined from similar events): error creating gateway for node worker-2.ostest.test.metalkube.org: failed to init shared interface gateway: error creating static route {UUID:70449439-0373-49c3-b544-1424e1d6e867 BFD:<nil> ExternalIDs:map[] IPPrefix:0.0.0.0/0 Nexthop:192.168.111.1 Options:map[] OutputPort:0xc000f37070 Policy:<nil> RouteTable:} in GR GR_worker-2.ostest.test.metalkube.org: unexpectedly found multiple results for provided predicate
event happened 135 times, something is wrong: node/worker-2.ostest.test.metalkube.org - reason/ErrorReconcilingNode roles/worker (combined from similar events): error creating gateway for node worker-2.ostest.test.metalkube.org: failed to init shared interface gateway: error creating static route {UUID:70449439-0373-49c3-b544-1424e1d6e867 BFD:<nil> ExternalIDs:map[] IPPrefix:0.0.0.0/0 Nexthop:192.168.111.1 Options:map[] OutputPort:0xc001df3010 Policy:<nil> RouteTable:} in GR GR_worker-2.ostest.test.metalkube.org: unexpectedly found multiple results for provided predicate
```
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ovn-kubernetes/1400/pull-ci-openshift-ovn-kubernetes-master-e2e-metal-ipi-ovn-dualstack/1597329218151124992

**Comment by trozet** [comment]
/hold

investigating failures:
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ovn-kubernetes/1400/pull-ci-openshift-ovn-kubernetes-master-e2e-metal-ipi-ovn-dualstack/1597396675079442432

File: go-controller/pkg/ovn/namespace.go [hunk 1]
```diff
@@ -231,7 +231,7 @@ func (oc *DefaultNetworkController) AddNamespace(ns *kapi.Namespace) error {
 
 	nsInfo, nsUnlock, err := oc.ensureNamespaceLocked(ns.Name, false, ns)
 	if err != nil {
-		errors = append(errors, fmt.Errorf("failed to ensure namespace locked: %v", err))
+		return fmt.Errorf("failed to ensure namespace locked: %v", err)
```
Comment (flavio-fernandes) [inline go-controller/pkg/ovn/namespace.go:234]: oops!

Comment (flavio-fernandes) [reply go-controller/pkg/ovn/namespace.go:234]: Looking at upstream https://github.com/ovn-org/ovn-kubernetes/commit/49f284eda11cad7cf986159f5d92496a39afca68#r91468230 I can see that this is only an issue DS due to the hybrid code that does not exist there.

**Review by flavio-fernandes — COMMENTED** [review]
/lgtm

**Pull request merged**